### ISSUE TITLE
ArgumentError: invalid JSON at byte position?

### ISSUE THREAD

**Issue by DataPsycho** [JuliaData/StructTypes.jl#13]
Hi, I was trying out the Example of StructType function given in the official document as follows:
```
struct CoolType
    val1::Int
    val2::Int
    val3::String
end

StructTypes.StructType(::Type{CoolType}) = StructTypes.Struct()

JSON3.read("{"val1": 1, "val2": 2, "val3": 3}", CoolType) 
JSON3.read("{"val2": 2, "val1": 1, "val3": 3}", CoolType) 
```
First of all compailer complain about `"{"val1": 1, "val2": 2, "val3": 3}"` being not a valid string and throws error as `ERROR: syntax: cannot juxtapose string literal` but this one `"{\"val2\": 2, \"val1\": 1, \"val3\": \"hey\"}"`  works. But it looks like the demonstration in the [doc string](https://github.com/JuliaData/StructTypes.jl/blob/56bfb9ee4a426aeabec083ece8f73b283f9def8a/src/StructTypes.jl#L40) given properly: 

Also putting numeric value instead of string in the field `JSON3.read("{\"val1\":1,\"val2\":2,\"val3\":3}", CoolType` does not work and throws the following error:
```
ERROR: ArgumentError: invalid JSON at byte position 27 while parsing type String
```
But the way the documents shows it should be working? `@assert JSON3.read("{"val2": 2, "val1": 1, "val3": 3}", CoolType) == CoolType(2, 1, "3")`

But the following works fine:
```
JSON3.read("{\"val2\":2,\"val1\":1,\"val3\":\"hey\"}", CoolType
```

**Comment by DilumAluthge** [comment]
Fixed by #17

**Comment by quinnj** [comment]
Yeah, sorry about the confusion; you need to escape quotes within strings in julia, like `" this quote \" is escaped"`. And the types do need to be strict; there's no automatic coercion.
